### PR TITLE
Quantile filter

### PR DESCRIPTION
**Pull request by konikvranik**
## Description:

Documents quantile filter

**Related issue (if applicable):** fixes esphome/feature-requests#1543

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#2900

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Review by jesserockz — APPROVED** [review]

File: components/sensor/index.rst [hunk 1]
```diff
@@ -232,6 +237,41 @@ degree with a least squares solver.
       filters:
         - filter_out: 85.0
 
+``quantile``
+************
+
+A `simple moving quantile <https://www.wikiwand.com/en/Quantile>`__
```
Comment (oxan) [inline components/sensor/index.rst]: Please link to the regular Wikipedia site, instead of a third-party copy.

File: components/sensor/index.rst [hunk 2]
```diff
@@ -232,6 +237,41 @@ degree with a least squares solver.
...
+      filters:
+        - quantile:
+            window_size: 7
+            send_every: 4
+            send_first_at: 3
+            quantile: .9
+
+Configuration variables:
+
+- **window_size** (*Optional*, int): The number of values over which to calculate the median
```
Comment (oxan) [inline components/sensor/index.rst]: Copy/paste error, median -> quantile.

File: components/sensor/index.rst [hunk 3]
```diff
@@ -232,6 +237,41 @@ degree with a least squares solver.
...
+  when pushing out a value. This number should
+  be odd if you want an actual received value pushed out.
```
Comment (oxan) [inline components/sensor/index.rst]: Copy/paste error, this doesn't apply to your filter.

File: components/sensor/index.rst [hunk 4]
```diff
@@ -232,6 +237,41 @@ degree with a least squares solver.
...
+  Defaults to ``5``.
+- **send_every** (*Optional*, int): How often a sensor value should be pushed out. For
+  example, in above configuration the median is calculated after every 4th
```
Comment (oxan) [inline components/sensor/index.rst]: Copy/paste error, median -> quantile.

**Review by oxan — CHANGES_REQUESTED** [review]

File: components/sensor/index.rst [hunk 5]
```diff
@@ -232,6 +237,41 @@ degree with a least squares solver.
...
+  Defaults to ``5``.
+- **send_every** (*Optional*, int): How often a sensor value should be pushed out. For
+  example, in above configuration the quantile is calculated after every 4th
+  received sensor value, over the last 7 received values.
+  Defaults to ``5``.
+- **send_first_at** (*Optional*, int): By default, the very first raw value on boot is immediately
+  published. With this parameter you can specify when the very first value is to be sent.
+  Must be smaller than or equal to ``send_every``
+  Defaults to ``1``.
+- **quantile** (**Optional**, float): value from 0 to 1 to determine which quantile to pick.
```
Comment (oxan) [inline components/sensor/index.rst]:
```suggestion
- **quantile** (*Optional*, float): value from 0 to 1 to determine which quantile to pick.
```

**Review by oxan — APPROVED** [review]

**Pull request merged**